src/cashier_frontend_new/src/modules/auth/state/auth.svelte.ts

```diff
@@ -90,6 +92,51 @@ export const authState = {
   get pnp() {
     return pnp;
   },
+  // Get anonymous agent
+  get anonAgent() {
+    if (!pnp) {
+      throw new Error("PNP is not initialized");
+    }
+    // create anonymous agent
+    return HttpAgent.createSync({
+      host: HOST_ICP,
+      shouldFetchRootKey: FEATURE_FLAGS.LOCAL_IDENTITY_PROVIDER_ENABLED,
+    });
+  },
+  // Generic method to get actor for any canister
+  // Using current identity if available, otherwise anonymous
```

Comment:
If an authenticated agent is required, this function should never silently return an anonymous one